utils.py

```diff
@@ -0,0 +1,152 @@
+import os
+import shutil
+import sys
+import configparser
+import datetime
+from pathlib import Path
+from itertools import product
+from tqdm import tqdm
+from uuid import uuid4
+import pandas as pd
+from process_results import create_run_csv
+from imagenet.model_saver import save_model
+
+
+import smartsim
+from smartsim import Experiment, status
+from smartredis import Client
+
+
+from smartsim.log import get_logger, log_to_file
+logger = get_logger("Scaling Tests")
+
+def get_date():
+    date = str(datetime.datetime.now().strftime("%Y-%m-%d"))
+    return date
+
+def _check_model(device, force_rebuild=False):
+    if device.startswith("GPU") and (force_rebuild or not Path("./imagenet/resnet50.GPU.pt").exists()):
+        from torch.cuda import is_available
+        if not is_available():
+            message = "resnet50.GPU.pt model missing in ./imagenet directory. \n"
+            message += "Since CUDA is not available on this node, the model cannot be created. \n"
+            message += "Please run 'python imagenet/model_saver.py --device GPU' on a node with an available CUDA device."
+            logger.error(message)
+            sys.exit(1)
+
+def create_folder(exp_name, launcher): 
+    i = 0
+    while os.path.exists("results/"+exp_name+"/run" + str(i)):
```

Comment:
I have the .join in use below but the .exists checks to see if the user basically deleted any of the runs and then begins there, now that I think about it, what could happen is number 7 is deleted out of 10 and then the program fills in number 7 then continues with 11, Ill have to add something for that. Maybe I should actually delete this implementation? Also, this might add unnecessary runtime?

Reply:
Oh, I should have been more clear...

I would just refactor this to something like
```py
import os.path as osp

osp.exists(osp.join("results", exp_name, "run", str(i)))
```
to avoid hard coding the path separator. Either that or format using `osp.sep`
```py
osp.exists(f"results{osp.sep}{exp_name}{osp.sep}run{osp.sep}{i}")
```
but I tend to find that harder to read

Reply:
Regarding the deletion of a previous-but-not-last-run-dir I don't think that's a huge deal, but if we do decide to refactor, maybe we could id it based on the time of the run to the second? Like `run-2023-04-10-2:59:33` or something?

Reply:
I like that idea so much!!

Reply:
I dont even need to check if it exists I believe so Im gonna completely get rid of the while block to check if it exists

Reply:
will this be an issue if the user runs the test from within a main file, like the cd thing we were talking about, Im thinking that the results folder might be created in the driverinference for example?

Reply:
I think the question above relates to what osp means, Im thinking that it just searches the path ur in, so if im in back smartsim-scaling/inferencedriver then itll make the results folder there instead of smartsim-scaling/